synapse/handlers/device.py

```diff
@@ -762,7 +762,7 @@ async def incoming_device_list_update(
         user_id = edu_content.pop("user_id")
         device_id = edu_content.pop("device_id")
         stream_id = str(edu_content.pop("stream_id"))  # They may come as ints
-        prev_ids = edu_content.pop("prev_id", [])
+        prev_ids = edu_content.pop("prev_id") or []
```

Comment:
Can we instead check if `prev_ids` is a list and if not raise a `SynapseError(400, "Device list update had an invalid 'prev_ids' field")`. This will get correctly handled by the caller.

Reply:
You got it, I've updated the PR